src/sentry/services/hybrid_cloud/actor/__init__.py

```diff
@@ -0,0 +1,62 @@
+# Please do not use
+#     from __future__ import annotations
+# in modules such as this one where hybrid cloud service classes and data models are
+# defined, because we want to reflect on type annotations and avoid forward references.
+
+import dataclasses
+from enum import Enum
+from typing import TYPE_CHECKING, Optional, Union
+
+from sentry.services.hybrid_cloud.user import RpcUser
+
+if TYPE_CHECKING:
+    from sentry.models import Team, User
+
+
+class ActorType(Enum):
+    USER = "User"
+    TEAM = "Team"
+
+
+@dataclasses.dataclass(frozen=True, eq=True)
+class RpcActor:
+    """Can represent any model object with a foreign key to Actor."""
+
+    id: int
+    actor_id: Optional[int]
+    actor_type: ActorType
+    is_superuser: bool = False
+
+    @classmethod
+    def from_object(cls, obj: Union["User", "Team", "RpcUser"]) -> "RpcActor":
+        from sentry.models import Team, User
+
+        if isinstance(obj, User):
+            return cls.from_orm_user(obj)
+        if isinstance(obj, Team):
```

Comment:
Could it also be `RpcTeam`?

Reply:
It won't be an RpcTeam in practice because the notifications logic runs in the region silo.